components/raftstore/src/store/async_io/write.rs

```diff
@@ -195,9 +199,20 @@ where
             extra_write: ExtraWrite::None,
             messages: vec![],
             trackers: vec![],
+            after_write_hook: None,
+            has_snapshot: false,
         }
     }
 
+    pub fn on_after_write_to_kv_db(&mut self) {
+        if self.after_write_hook.is_some() {
+            let hook = std::mem::take(&mut self.after_write_hook).unwrap();
+            if let Err(e) = hook() {
```

Comment:
The hook should not return any error. async writer has no idea what's the proper way to handle.

Reply:
fix